app/src/main/java/org/schabi/newpipe/CheckForNewAppVersionTask.java

```diff
@@ -0,0 +1,162 @@
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * AsyncTask to check if there is a newer version of the NewPipe github apk available or not.
+ * If there is a newer version we show a notification, informing the user. On tapping
+ * the notification, the user will be directed to the download link.
+ */
+public class CheckForNewAppVersionTask extends AsyncTask<Void, Void, String> {
+
+    private Application app = App.getContext();
+
+    private String newPipeApiUrl = "https://newpipe.schabi.org/api/data.json";
+    private int timeoutPeriod = 10000;
+
+    @Override
+    protected void onPreExecute() {
+        // Continue with version check only if the build variant is of type "github".
+        if (!BuildConfig.FLAVOR.equals(app.getString(R.string.app_flavor_github))) {
+            // this.cancel(true);
+        }
+    }
+
+    @Override
+    protected String doInBackground(Void... voids) {
+
+        // Make a network request to get latest NewPipe data.
+
+        String response;
+        HttpURLConnection connection = null;
+
+        try {
+
+            URL url = new URL(newPipeApiUrl);
+
+            connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+            connection.setConnectTimeout(timeoutPeriod);
+            connection.setReadTimeout(timeoutPeriod);
+            connection.setRequestProperty("Content-length", "0");
+            connection.setUseCaches(false);
+            connection.setAllowUserInteraction(false);
+            connection.connect();
+
+            int responseStatus = connection.getResponseCode();
+
+            switch (responseStatus) {
+
+                case 200:
+                case 201:
+                    BufferedReader bufferedReader = new BufferedReader(
+                            new InputStreamReader(connection.getInputStream()));
+
+                    StringBuilder stringBuilder = new StringBuilder();
+
+                    String line;
+
+                    while ((line = bufferedReader.readLine()) != null) {
+                        stringBuilder.append(line);
+                        stringBuilder.append("\n");
+                    }
+
+                    bufferedReader.close();
+                    response = stringBuilder.toString();
+
+                    return response;
+            }
+        } catch (MalformedURLException ex) {
+            ex.printStackTrace();
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        } finally {
+            if (connection != null) {
+                try {
+                    connection.disconnect();
+                } catch (Exception ex) {
+                    ex.printStackTrace();
+                }
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    protected void onPostExecute(String response) {
+
+        Log.i("Response--", response);
```

Comment:
This entry does not help us to find issues in the future. Please add some additional information or  remove this log entry.